nuxeo-mqueues-core/src/main/java/org/nuxeo/ecm/platform/importer/mqueues/mqueues/chronicle/ChronicleMQueue.java

```diff
@@ -207,5 +269,50 @@ private int findNbQueues(File basePath) {
         return ret;
     }
 
+    @Override
+    public void onAcquired(int cycle, File file) {
+
+        log.debug("New file: " + file + " cycle: " + cycle);
+
+        // Retrieve queue
+        String queueFileName = file.getParentFile().getName();
+        int queueIndex = Integer.valueOf(queueFileName.substring(queueFileName.length() - 2));
+        SingleChronicleQueue queue = (SingleChronicleQueue) queues.get(queueIndex);
+
+        RollCycle rollCycle = queue.rollCycle();
+        long epoch = queue.epoch();
+
+        RollingResourcesCache cache = new RollingResourcesCache(rollCycle, epoch,
+                name -> new File(queue.file().getAbsolutePath(), name + SUFFIX),
+                f -> f.getName().substring(0, f.getName().length() - SUFFIX.length()));
+
+        final TimeUnit timeUnit;
+        if (RollCycles.TEST_SECONDLY.equals(rollCycle)) {
+            timeUnit = TimeUnit.SECONDS;
+        } else if (RollCycles.MINUTELY.equals(rollCycle)) {
+            timeUnit = TimeUnit.MINUTES;
+        } else if (RollCycles.HOURLY.equals(rollCycle)) {
+            timeUnit = TimeUnit.HOURS;
+        } else {
+            timeUnit = TimeUnit.DAYS;
+        }
+
+        int firstCycle = queue.firstCycle();
+        int lastCycle = rollCycle.current(() -> timeUnit.toMillis(cycle - retentionDuration), epoch);
+
+        // Delete old files in queue
+        String[] files = queue.file().list();
+        if (files != null && firstCycle < lastCycle) {
+            Arrays.stream(files)
+                  .map(fname -> fname.substring(0, fname.length() - SUFFIX.length()))
+                  .map(cache::parseCount)
+                  .filter(c -> c < lastCycle)
+                  .forEach(c -> cache.resourceFor(c).path.delete());
+        }
```

Comment:
wow too long and too complex, you should just filter a list of file, then apply a delete with a log.info with absolute path of the file you remove.